### PR TITLE
Update firebase-hosting.md

### PR DESCRIPTION
**Pull request by himesh-cm**
I added documentation on deploying Firebase hosting using a service account user.

### CODE REVIEW

File: content/yaml-publishing/firebase-hosting.md [hunk 1]
```diff
@@ -11,6 +11,45 @@ weight: 7
...
 
+
+### Using Firebase Google Service Account
+1. To get started you will need a Google service account user with `Cloud Build Service Account`, `Firebase Admin` and `API Keys Admin roles` as shown in the Firebase [docs](https://cloud.google.com/build/docs/deploying-builds/deploy-firebase#required_iam_permissions).
+
+2. You can follow the step-by-step guide to create the service account user [here](../yaml-publishing/firebase-app-distribution/)
+
+3. Configure the `JSON` key received from step 2 as an environment variable name `FIREBASE_SERVICE_ACCOUNT` .
+
+4. Add the path where the contents of the `JSON` key will be copied to as an environment variable name `GOOGLE_APPLICATION_CREDENTIALS` and the value as `$CM_BUILD_DIR/firebase_credentials.json`.
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:22]: Sounds a bit confusing, perhaps sth like: Add an environment variable name `GOOGLE_APPLICATION_CREDENTIALS` and set its the value to `$CM_BUILD_DIR/firebase_credentials.json`

File: content/yaml-publishing/firebase-hosting.md [hunk 2]
```diff
@@ -11,6 +11,45 @@ weight: 7
...
+
+5. Make sure to mark it as `secure` and Enter the variable group name, e.g. **_firebase_credentials_**.
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:24]: Perhaps add a link to the page where env var groups are explained

File: content/yaml-publishing/firebase-hosting.md [hunk 3]
```diff
@@ -11,6 +11,45 @@ weight: 7
...
+
+6. In your workflow use the below script to copy the contents of the `JSON` key in your project folder.
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:26]: 6. In your workflow, use the below script to copy the contents of the `JSON` key in your project folder.

File: content/yaml-publishing/firebase-hosting.md [hunk 4]
```diff
@@ -11,6 +11,45 @@ weight: 7
...
+{{< highlight yaml "style=paraiso-dark">}}
+scripts:
+  - name: Write Google credentials
+    script: | 
+      echo $FIREBASE_SERVICE_ACCOUNT > $GOOGLE_APPLICATION_CREDENTIALS
+
+{{< /highlight >}}
+
+
+7. Create a new script for publishing to Firebase Hosting in your scripts section of the `.yaml` file and add it right after the build step. Make sure to import the `firebase_credentials` env group in your `environment` section as shown below.
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:38]: Step 5 already mentions about adding creds, you can remove that part together with its sample below and let's update the 1st part to sth like this: For publishing to Firebase Hosting, add the following script in codemagic.yaml right after executing the build script:

**Review by icarusdust — CHANGES_REQUESTED** [review]

File: content/yaml-publishing/firebase-hosting.md (hunk 1 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md:22]: ✅

File: content/yaml-publishing/firebase-hosting.md (hunk 3 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md:26]: ✅

File: content/yaml-publishing/firebase-hosting.md (hunk 4 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md:38]: ✅

File: content/yaml-publishing/firebase-hosting.md (hunk 2 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md:24]: i think it will get confusing. Already explained in short how to add here.

**Comment by himesh-cm** [comment]
@icarusdust suggested changes implemented; please review.

File: content/yaml-publishing/firebase-hosting.md (hunk 2 above)
Comment (icarusdust) [reply content/yaml-publishing/firebase-hosting.md:24]: You are not explaining where the secure checkbox is, what a group is and how to reference it etc. Let's please add a link to that page.

**Review by icarusdust — CHANGES_REQUESTED** [review]

File: content/yaml-publishing/firebase-hosting.md (hunk 2 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md]: ✅ done please review
@icarusdust

File: content/yaml-publishing/firebase-hosting.md [hunk 5]
```diff
@@ -11,6 +11,45 @@ weight: 7
...
+### Using Firebase Google Service Account
+1. To get started you will need a Google service account user with `Cloud Build Service Account`, `Firebase Admin` and `API Keys Admin roles` as shown in the Firebase [docs](https://cloud.google.com/build/docs/deploying-builds/deploy-firebase#required_iam_permissions).
+
+2. You can follow the step-by-step guide to create the service account user [here](../yaml-publishing/firebase-app-distribution/).
+
+3. Configure the `JSON` key received from step 2 as an environment variable name `FIREBASE_SERVICE_ACCOUNT` .
+
+4. Add an environment variable name GOOGLE_APPLICATION_CREDENTIALS and set its the value to `$CM_BUILD_DIR/firebase_credentials.json`.
+
+5. Make sure to mark it as `secure` and Enter the variable group name, e.g. **_firebase_credentials_**. You can check the doc [here](../yaml-basic-configuration/configuring-environment-variables/) for more details.
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:24]: maybe add this to step 4, not as a separate step.

File: content/yaml-publishing/firebase-hosting.md [hunk 6]
```diff
@@ -11,6 +11,45 @@ weight: 7
...
+    script: | 
+      echo $FIREBASE_SERVICE_ACCOUNT > $GOOGLE_APPLICATION_CREDENTIALS
+
+{{< /highlight >}}
+
+
+7. For publishing to Firebase Hosting, add the following script in codemagic.yaml right after executing the build script.
+
+{{< highlight yaml "style=paraiso-dark">}}
+environment:
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:41]: remove this part please

**Review by icarusdust — CHANGES_REQUESTED** [review]

File: content/yaml-publishing/firebase-hosting.md (hunk 5 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md]: ✅

File: content/yaml-publishing/firebase-hosting.md (hunk 6 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md]: ✅

File: content/yaml-publishing/firebase-hosting.md [hunk 7]
```diff
@@ -11,6 +11,38 @@ weight: 7
...
+
+{{< highlight yaml "style=paraiso-dark">}}
+scripts:
+  - name: Write Google credentials
+    script: | 
+      echo $FIREBASE_SERVICE_ACCOUNT > $GOOGLE_APPLICATION_CREDENTIALS
+{{< /highlight >}}
+
+
+6. For publishing to Firebase Hosting, add the following script in codemagic.yaml right after executing the build script.
```
Comment (icarusdust) [inline content/yaml-publishing/firebase-hosting.md:35]: last one, GH removes it when adding it as a comment, put star signs around codemagic.yaml

**Review by icarusdust — CHANGES_REQUESTED** [review]

File: content/yaml-publishing/firebase-hosting.md (hunk 7 above)
Comment (himesh-cm) [reply content/yaml-publishing/firebase-hosting.md]: ✅

**Review by icarusdust — APPROVED** [review]
LGTM

**Pull request merged**